client/express-checkout/index.js

```diff
@@ -441,6 +441,18 @@ jQuery( ( $ ) => {
 
 					$.when( wcpayECE.getSelectedProductData() )
 						.then( ( response ) => {
+							// We do not support variable subscriptions with variations
+							// that require shipping and include a free trial.
+							if (
+								getExpressCheckoutData( 'product' )
+									.product_type === 'variable-subscription' &&
```

Comment:
There's probably no harm in checking this, but just for my understanding given that this is happening within the `woocommerce_variation_has_changed` event, is it safe to assume the product type or is there a case where that assumption would be wrong?

Reply:
@bborman22 the product can also be a variable product, not necessarily a variable subscription.

Reply:
Oh, maybe you meant that only subscriptions can have free trials? If so, then yes.